src/Log/LogService.php

```diff
@@ -424,6 +424,25 @@ public function getDefaultFormat()
         return $format;
     }
 
+    /**
+     * Returns custom log format
+     *
+     * @param $logName string
+     * @return string
+     */
+    public function getCustomLogFormat($logName)
```

Comment:
This would seem to solve the current problem, however there are a couple of things missing in the long run. This does not allow using `%runId%` in formats provided as a parameter when creating a log. Also we have now two separate functions that read the format from the configuration. Two places that we will have to be maintained if there is a change. There are two aspects to this. Where do I get the format string? Either from `getLogConfig` or as a parameter. That is fine. Preparing the format string by replacing %runId% is an independent step and mixing those two things is the problem in the old code. You need get rid of the new function here. Instead add one that prepares format strings and use it in the right spot.